### PR TITLE
latest version of maven-javadoc-plugin

### PR DESCRIPTION
**Pull request by honnix**
# Hey, I just made a Pull Request!

## Description

Use latest version of maven-javadoc-plugin with a few tweaks.

The javadoc executable now lives in ${java.home}/bin/javadoc .

## Checklist for PR author(s)

- [ ] Changes are covered by unit test
- [ ] All tests pass
- [ ] Code coverage check passes
- [ ] Error handling is tested
- [ ] Errors are handled at the appropriate layer
- [ ] Errors that cannot be handled where they occur are propagated
- [ ] (optional) Changes are covered by system test
- [ ] Relevant documentation updated
- [ ] This PR has NO breaking change to public API
- [ ] This PR has breaking change to public API and it is documented

## Checklist for PR reviewer(s)

- [ ] This PR has been incorporated in release note for the coming version
- [ ] Risky changes introduced by this PR have been all considered

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -618,6 +618,26 @@
...
             </plugin>
+            <plugin>
+              <artifactId>maven-javadoc-plugin</artifactId>
+              <version>3.0.1</version>
+              <configuration>
+                <additionalparam>-Xdoclint:none</additionalparam>
+                <sourceFileExcludes>
+                  <sourceFileExclude>**/target/generated-sources/**</sourceFileExclude>
+                </sourceFileExcludes>
+                <javadocExecutable>${java.home}/bin/javadoc</javadocExecutable>
```
Comment (danielnorberg) [inline pom.xml]: 🤔 Afaict this shouldn't be necessary

Comment (honnix) [reply pom.xml]: I will send you the internal stackoverflow explaining this.

File: pom.xml [hunk 2]
```diff
@@ -618,6 +618,26 @@
                 </execution>
               </executions>
             </plugin>
+            <plugin>
+              <artifactId>maven-javadoc-plugin</artifactId>
+              <version>3.0.1</version>
+              <configuration>
+                <additionalparam>-Xdoclint:none</additionalparam>
```
Comment (danielnorberg) [inline pom.xml]: Not sure this parameter is still there, might have been replaced in `3.x`: https://maven.apache.org/plugins/maven-javadoc-plugin/jar-mojo.html#additionalOptions

Comment (honnix) [reply pom.xml]: I will remove this and see what happens.

**Pull request merged**